Ginger/GingerCoreCommon/Repository/RepositoryFolderBase.cs

```diff
@@ -138,5 +140,33 @@ public virtual void ResumeFileWatcher()
         /// </summary>
         /// <param name="repositoryFolder"></param>
         public abstract void MoveItem(RepositoryItemBase repositoryItem, RepositoryFolderBase repositoryFolder);
+             
+        private static ISourceControl SourceControl;
+        
```

Comment:
please do not do actual icon refresh inside the RepositoryFolder, all of this refresh need to be done in the TreeItem node and not here because the tree needs it and not the Folder